cli/install/install.go

```diff
@@ -536,16 +536,68 @@ func installTt(binDir string, installCtx InstallCtx, distfiles string) error {
 	}
 
 	// Check if that version is already installed.
+	// If it is installed, check if the newest version exists.
 	if !installCtx.Reinstall {
 		log.Infof("Checking existing...")
 		if util.IsRegularFile(filepath.Join(binDir, versionStr)) {
-			log.Infof("%s version of tt already exists, updating symlink...", versionStr)
-			err := util.CreateSymlink(versionStr, filepath.Join(binDir, "tt"), true)
+			isBinExecutable, err := util.IsExecOwner(filepath.Join(binDir, versionStr))
 			if err != nil {
 				return err
 			}
-			log.Infof("Done")
-			return err
+
+			var curBinVersion, lastCommitHash string
+
+			// We need to make sure that we check newest commits only for
+			// production 'master' branch. Also we want to ask if user wants
+			// to check for updates.
+			var answer bool
+			if isBinExecutable && ttVersion == "master" {
+				var err error
+				answer, err = util.AskConfirm(os.Stdin, "You already have installed tt_master.\n"+
+					"Want to check for newest versions?")
+				if err != nil {
+					return err
+				}
+			}
+
+			if answer {
+				lastCommitHash, err = getCommit(installCtx.Local, distfiles,
+					search.ProgramTt, ttVersion)
+				if err != nil {
+					return err
+				}
+
+				binVersion := version.GetVersion(false, true)
+				// binVersion returns string like '<x.x.x>.<hash>'.
+				// We need to divide hash and version using this dot separator.
+				sepIndex := strings.LastIndex(binVersion, ".")
+				if sepIndex == -1 {
+					return fmt.Errorf("couldn't parse version of current tt binary")
+				}
+				if len(binVersion) < sepIndex+1 {
+					return fmt.Errorf("couldn't check version of an uninstalled tt")
+				}
+				curBinVersion = binVersion[sepIndex+1:]
+			}
```

Comment:
It will be more clear to have the branch in the place where it is relevant.
```suggestion
				if answer {
					lastCommitHash, err = getCommit(installCtx.Local, distfiles,
						search.ProgramTt, ttVersion)
					if err != nil {
						return err
					}

					binVersion := version.GetVersion(false, true)
					// binVersion returns string like '<x.x.x>.<hash>'.
					// We need to divide hash and version using this dot separator.
					sepIndex := strings.LastIndex(binVersion, ".")
					if sepIndex == -1 {
						return fmt.Errorf("couldn't parse version of current tt binary")
					}
					if len(binVersion) < sepIndex+1 {
						return fmt.Errorf("couldn't check version of an uninstalled tt")
					}
					curBinVersion = binVersion[sepIndex+1:]
				}
			}
```

Reply:
Moved this logic to separate function.